targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/i2c_api.c

```diff
@@ -0,0 +1,216 @@
+    pinmap_pinout(sda, PinMap_I2C_SDA);
+    pinmap_pinout(scl, PinMap_I2C_SCL);
+    SystemCoreClockUpdate();
+    I2C_Return = adi_i2c_Open(I2C_DevNum, I2C_Mem, ADI_I2C_MEMORY_SIZE, pI2C_Handle);
+    if (I2C_Return) {
+        obj->error = I2C_EVENT_ERROR;
+        return;
+    }
+    I2C_Return = adi_i2c_Reset(*pI2C_Handle);
+    if (I2C_Return) {
+        obj->error = I2C_EVENT_ERROR;
+        return;
+    }
+}
+
+
+int i2c_start(i2c_t *obj)
+{
+    return 0;
+}
+
+
+int i2c_stop(i2c_t *obj)
+{
+    return 0;
+}
+
+
+void i2c_frequency(i2c_t *obj, int hz)
+{
+    ADI_I2C_HANDLE  I2C_Handle;
+    ADI_I2C_RESULT  I2C_Return = ADI_I2C_SUCCESS;
+
+
+    I2C_Handle = *obj->pI2C_Handle;
+    I2C_Return = adi_i2c_SetBitRate(I2C_Handle, (uint32_t) hz);
+    if (I2C_Return) {
+        obj->error = I2C_EVENT_ERROR;
+        return;
+    }
+}
+
+
+int i2c_read(i2c_t *obj, int address, char *data, int length, int stop)
+{
+    ADI_I2C_RESULT      I2C_Return;
+    ADI_I2C_TRANSACTION I2C_inst;
+    uint8_t             I2C_PrologueData = 0x00;
+    uint32_t            I2C_Errors;                 /* HW Error result */
+    ADI_I2C_HANDLE      I2C_Handle;
+
+
+    I2C_Handle = *obj->pI2C_Handle;
+    I2C_Return = adi_i2c_SetSlaveAddress(I2C_Handle, (address & 0x0000FFFF));
+    I2C_inst.pPrologue     = &I2C_PrologueData;
+    I2C_inst.nPrologueSize = 0;
+    I2C_inst.pData         = (uint8_t*) data;
+    I2C_inst.nDataSize     = length;
+    I2C_inst.bReadNotWrite = true;
+    I2C_inst.bRepeatStart  = stop;
+    I2C_Return = adi_i2c_ReadWrite(I2C_Handle, &I2C_inst, &I2C_Errors);
+    if (I2C_Return) {
+        obj->error = I2C_EVENT_ERROR;
+        return -1;
+    } else {
+        return length;
+    }
+}
+
+
+int i2c_write(i2c_t *obj, int address, const char *data, int length, int stop)
+{
+    ADI_I2C_RESULT      I2C_Return = ADI_I2C_SUCCESS;
+    ADI_I2C_TRANSACTION I2C_inst;
+    uint8_t             I2C_PrologueData = 0x00;
+    uint32_t            I2C_Errors;                 /* HW Error result */
+    ADI_I2C_HANDLE      I2C_Handle;
+
+
+    I2C_Handle = *obj->pI2C_Handle;
+    I2C_Return = adi_i2c_SetSlaveAddress(I2C_Handle, (address & 0x0000FFFF));
+    I2C_inst.pPrologue      = &I2C_PrologueData;
+    I2C_inst.nPrologueSize  = 0;
+    I2C_inst.pData          = (uint8_t*) data;
+    I2C_inst.nDataSize      = length;
+    I2C_inst.bReadNotWrite  = false;
+    I2C_inst.bRepeatStart   = stop;
+    I2C_Return = adi_i2c_ReadWrite(I2C_Handle, &I2C_inst, &I2C_Errors);
+    if (I2C_Return) {
+        obj->error = I2C_EVENT_ERROR;
+        return -1;
+    } else {
+        return length;
+    }
+}
+
+
+void i2c_reset(i2c_t *obj)
+{
+    ADI_I2C_RESULT      I2C_Return;
+    ADI_I2C_HANDLE      I2C_Handle = *obj->pI2C_Handle;
+
+    I2C_Return = adi_i2c_Reset(I2C_Handle);
+    if (I2C_Return) {
+        obj->error = I2C_EVENT_ERROR;
+        return;
+    }
+}
+
+
+int i2c_byte_read(i2c_t *obj, int last)
+{
+    return 0;
```

Comment:
Is this limitation of hw or yet not implemented? Please provide more information. Leaving any API empty should have a reasoning .